File: bestia-messages/src/main/java/net/bestia/messages/inventory/InventoryItemCastConfirmMessage.java
package net.bestia.messages.inventory;

import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * The server confirms the casting of an item to the client.
 * 
 * @author Thomas Felix <thomas.felix@tfelix.de>
 *
 */
public class InventoryItemCastConfirmMessage extends InventoryItemCastMessage {

	private static final long serialVersionUID = 1L;
	
	public static final String MESSAGE_ID = "inventory.item.castconfirm";
	
	@JsonProperty("s")
	private boolean success = true;
	
	
	public InventoryItemCastConfirmMessage(InventoryItemCastMessage msg, boolean success) {
		super(msg.getAccountId(), msg);
		
		this.success = success;
	}

	@Override
	public String getMessageId() {
		return MESSAGE_ID;
	}
	
	public boolean getSuccess() {
		return this.success;
	}
	
	public void setSuccess(boolean success) {
		this.success = success;
	}
	
	@Override
	public String toString() {
		return String.format("InventoryItemCastConfirmMessage[success: %b]", success);
	}
}

File: bestia-messages/src/main/kotlin/net/bestia/messages/inventory/InventoryItemCastConfirmMessage.kt
package net.bestia.messages.inventory

import com.fasterxml.jackson.annotation.JsonProperty
import net.bestia.messages.AccountMessage
import net.bestia.messages.MessageId

/**
 * The server confirms the casting of an item to the client.
 *
 * @author Thomas Felix
 */
data class InventoryItemCastConfirmMessage(
    override val accountId: Long,

    @JsonProperty("s")
    val success: Boolean,

    @JsonProperty("t")
    val token: String
) : AccountMessage, MessageId {

  override val messageId: String
    get() = MESSAGE_ID

  companion object {
    const val MESSAGE_ID = "inventory.item.castconfirm"
  }
}
